oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/PipelinedMergeSortTaskTest.java

```diff
@@ -89,27 +94,41 @@ private PipelinedMergeSortTask.Result runTest(Compression algorithm, Path... fil
         Path sortRoot = sortFolder.getRoot().toPath();
         // +1 for the Sentinel.
         ArrayBlockingQueue<Path> sortedFilesQueue = new ArrayBlockingQueue<>(files.length + 1);
-        PipelinedMergeSortTask mergeSortTask = new PipelinedMergeSortTask(sortRoot, pathComparator, algorithm, sortedFilesQueue);
-        // Enqueue all the files that are to be merged
-        for (Path file : files) {
-            // The intermediate files are deleted after being merged, so we should copy them to the temporary sort root folder
-            Path workDirCopy = Files.copy(file, sortRoot.resolve(file.getFileName()));
-            sortedFilesQueue.put(workDirCopy);
+        ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+        try (MetricStatisticsProvider metricStatisticsProvider = new MetricStatisticsProvider(null, executor)) {
```

Comment:
Since `null` is passed into the `MetricStatisticsProvider`, I think it would be nice to know why that is, instead of `ManagementFactory.getPlatformMBeanServer()`?

Reply:
This constructor is called in many tests with null, I don't want to spread the same comment in so many places. The constructor should have a comment explaining that null is accepted and what that means (it does not, but that's a separate issue). And from the comments that should be in the constructor, it would be clear why we are passing null (i.e., the test does not need an MBean server).

Reply:
Is that because you are not interested in testing whether the values of the metrics? To me it's not 100 % clear why the tests don't need an MBean server - in other tests, I see they are there. E.g. I remember it was necessary in the tests for the `NodeCounter` a while ago.

Reply:
If we pass null to the constructor, the `MetricStatisticsProvider` will use the `ManagementFactory.getPlatformMBeanServer()`. I think the problem here is lack of documentation on the construtor of `MetricStatisticsProvider`. I don't think this PR should correct the missing documentation.